tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1

```diff
@@ -213,351 +207,397 @@ param(
+    }
+    if ([string]::IsNullOrEmpty($aideuserConfig.Azure.Location)) {
+        throw "Require Location for Azure Arc"
+    } elseif ($arcLocations -inotcontains $aideuserConfig.Azure.Location) {
+        Write-Host "Supported Locations : $arcLocations"
+        throw "Location $($aideuserConfig.Azure.Location) is not supported for Azure Arc"
+    }
+    if ([string]::IsNullOrEmpty($aksedgeConfig.Arc.ClusterName)) {
+        throw "Require ClusterName for Azure Arc" 
+    }
 }
 
-# Here string for the json content
-$aideuserConfig = @"
+function Get-AkseeInstalledProductName
 {
-    "SchemaVersion": "1.1",
-    "Version": "1.0",
-    "AksEdgeProduct": "$productName",
-    "AksEdgeProductUrl": "$productUrl",
-    "Azure": {
-        "SubscriptionName": "",
-        "SubscriptionId": "$SubscriptionId",
-        "TenantId": "$TenantId",
-        "ResourceGroupName": "$ResourceGroupName",
-        "ServicePrincipalName": "aksedge-sp",
-        "Location": "$Location",
-        "CustomLocationOID":"$CustomLocationOid",
-        "EnableWorkloadIdentity": true,
-        "EnableKeyManagement": true,
-        "GatewayResourceId": "",
-        "Auth":{
-            "ServicePrincipalId":"",
-            "Password":""
-        }
-    },
-    "AksEdgeConfigFile": "aksedge-config.json"
+
+    return (Get-ChildItem -Path 'HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\' | Get-ItemProperty |  Where-Object {$_.DisplayName -like "*Aks Edge Essentials*"}).DisplayName
 }
-"@
-$aksedgeConfig = @"
+
+function EnsurePreRequisites
 {
-    "SchemaVersion": "1.15",
-    "Version": "1.0",
-    "DeploymentType": "SingleMachineCluster",
-    "Init": {
-        "ServiceIPRangeSize": 10
-    },
-    "Network": {
-        "NetworkPlugin": "$networkplugin",
-        "InternetDisabled": false,
-        "Proxy": {
-            "Http": null,
-            "Https": null,
-            "No": null
-        }
-    },
-    "User": {
-        "AcceptEula": true,
-        "AcceptOptionalTelemetry": true
-    },
-    "Machines": [
-        {
-            "LinuxNode": {
-                "CpuCount": 4,
-                "MemoryInMB": 10240,
-                "DataSizeInGB": 40,
-                "LogSizeInGB": 4
-            }
+
+    if (! [Environment]::Is64BitProcess) {
+        throw "Error: Run this in 64bit Powershell session"
+    }
+
+    # Validate az cli version.
+    $azVersion = (az version)[1].Split(":")[1].Split('"')[1]
+    $azMinRequiredVersion = "2.64.0"
+    if ($azVersion -lt $azMinRequiredVersion){
+        throw "Installed Azure CLI version $azVersion is older than $azMinRequiredVersion. Please upgrade Azure CLI and retry."
+    }
+
+    $installedAkseeProductName = Get-AkseeInstalledProductName
+    if (-Not [string]::IsNullOrEmpty($installedAkseeProductName)) {
+        if ($installedAkseeProductName -like "*K8s*") {
+            Write-Host "Detected AKSEE k8s installation. Please uninstall and run the script again!" -ForegroundColor Red
+            exit -1
```

Comment:
do we want to replace all `exit -1` with `throw` in this file?

Reply:
fix

Reply:
will do the rest in a separate PR...